### PR TITLE
Better array

### PR DESCRIPTION
**Pull request by ricardoquesada**
Adds support for `std::vector + RCPtr<Object>` in `CCArray`.

Why would we want it ?
Basically it allow us to make performance tests on CCArray.

Last, but not least, the sideeffect is that now `CCArray` has a cleaner API.

ah... additionally all getters with the `get`, were deprecated, and the `get` version was added.

### CODE REVIEW

File: samples/Cpp/TestCpp/Classes/SchedulerTest/SchedulerTest.cpp [hunk 1]
```diff
@@ -695,11 +695,11 @@ void SchedulerUpdate::onEnter()
 void SchedulerUpdate::removeUpdates(float dt)
 {
     auto children = getChildren();
-    Node* node;
 
     for (auto c : *children)
     {
-        node = static_cast<Node*>(c);
+        auto obj = static_cast<Object*>(c);
```
Comment (minggo) [inline samples/Cpp/TestCpp/Classes/SchedulerTest/SchedulerTest.cpp:701]: Don't need to convert to `Object*`.

File: cocos2dx/touch_dispatcher/CCTouchDispatcher.cpp [hunk 2]
```diff
@@ -290,9 +304,10 @@ TouchHandler* TouchDispatcher::findHandler(Array* pArray, TouchDelegate *pDelega
     return NULL;
 }
 
-void TouchDispatcher::rearrangeHandlers(Array *pArray)
+void TouchDispatcher::rearrangeHandlers(Array *array)
 {
-    std::sort(pArray->data->arr, pArray->data->arr + pArray->data->num, less);
+    std::sort(array->data->arr, array->data->arr + array->data->num, less);
```
Comment (minggo) [inline cocos2dx/touch_dispatcher/CCTouchDispatcher.cpp:309]: So can only work when using ccArray in Array?

Comment (ricardoquesada) [inline cocos2dx/touch_dispatcher/CCTouchDispatcher.cpp:309]: Apparently the iterator that you added for `Array` doesn't work with

``` c++
std::sort( std::begin(*array), std::end(*array), less);
```

Not sure why :-( ... but that is something that needs to be fixed.

Comment (minggo) [inline cocos2dx/touch_dispatcher/CCTouchDispatcher.cpp:309]: Ok, i will fix it after merging.

**Pull request merged**